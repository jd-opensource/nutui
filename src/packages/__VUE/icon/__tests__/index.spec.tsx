import { mount } from '@vue/test-utils';
import { Add, IconFont } from '@nutui/icons-vue';

test('Icon: svg color width height', async () => {
  const wrapper = mount(() => {
    return <Add color="pink" width={30} height={30} />;
  });

  const svg = wrapper.find('.nut-icon');
  expect(svg.classes()).toContain('nut-icon-add');
  expect(svg.attributes('style')).toContain('color: pink');
  expect(svg.attributes('style')).toContain('height: 30px');
  expect(svg.attributes('style')).toContain('width: 30px');
});

test('Icon: icofnont color width height', async () => {
  const wrapper = mount(() => {
    return <IconFont name="add" color="pink" width={30} height={30} />;
  });

  const svg = wrapper.find('.nut-icon');
  expect(svg.classes()).toContain('nut-icon-add');
  expect(svg.attributes('style')).toContain('color: pink');
  expect(svg.attributes('style')).toContain('height: 30px');
  expect(svg.attributes('style')).toContain('width: 30px');
});
