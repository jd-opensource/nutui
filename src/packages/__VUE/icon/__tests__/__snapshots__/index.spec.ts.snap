// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render icon dongdong 1`] = `""`;

exports[`should render icon dongdong color 1`] = `"<i class=\\"nutui-iconfont nut-icon nut-icon-dongdong\\" style=\\"color: red;\\" src=\\"\\"></i>"`;

exports[`should render icon dongdong size 1`] = `"<i class=\\"nutui-iconfont nut-icon nut-icon-dongdong\\" style=\\"font-size: 24px; width: 24px; height: 24px;\\" src=\\"\\"></i>"`;

exports[`should render icon image type 1`] = `""`;
